setuptools_rust/setuptools_ext.py

```diff
@@ -163,13 +163,18 @@ def run(self) -> None:
                 build_rust.inplace = self.inplace
                 build_rust.target = self.target
                 build_rust.verbose = self.verbose
-                options = self.distribution.get_cmdline_options().get("bdist_wheel", {})
-                plat_name = options.get("plat-name") or self.plat_name
-                build_rust.plat_name = plat_name
+                build_rust.plat_name = self._get_wheel_plat_name() or self.plat_name
                 build_rust.run()
 
             build_ext_base_class.run(self)
 
+        def _get_wheel_plat_name(self) -> Optional[str]:
+            try:
+                cmd = self.distribution.get_command_obj("bdist_wheel")
+                return cast(Optional[str], cmd.plat_name)
+            except Exception:  # unlikely scenario: `wheel` not installed
+                return None
```

Comment:
For this specific scenario skipping `ensure_finalized` should be fine because `build_rust` already implements the same fallback for `plat_name` than `bdist_wheel` does (i.e. copying from `build`), so we are just interested in the value given via `sys.argv`/config file, which is available right after setuptools instantiates the command object inside `get_command_obj`.

But I agree that it would be nicer to share the logic between these 2 objects, so I will go ahead and implement it once I know what to do with the problem in auditwheel.